### ISSUE TITLE
MusicMagic always uses 1st track of seeds for DSTM

### ISSUE THREAD

**Issue by CDrummond** [LMS-Community/slimserver#372]
The MusicMagic DSTM plugin always uses the first track of a set of seeds, and asks for mixes based upon that for each seed track. Lines 334-335 of `Slim/Plugin/MusicMagic/Plugin.pm` have:

```
foreach (@$seedTracks) {
    my ($trackObj) = Slim::Schema->find('Track', $seedTracks->[0]->{id});
```

perhaps this should be:
```
foreach my $seedTrack (@$seedTracks) {
    my ($trackObj) = Slim::Schema->find('Track', $seedTrack->{id});
```

--------------------------------------------
Edit: Also, a call to MusicIP is made for each seed track (although with this bug its the same track for each seed track). However, you could make 1 MusicIP call with all seed tracks. I've create for DSTM only fork of this plugin where I have implemented 1 MIP call with all tracks. See: https://github.com/CDrummond/lms-mipmixer

**Comment by michaelherger** [comment]
Why don't you provide a PR for the plugin? Would be easier than trying to track down your changes...

**Comment by CDrummond** [comment]
Done - https://github.com/Logitech/slimserver/pull/373

But, the main point was the two line fix above.

**Comment by CDrummond** [comment]
Closing this as fixed by #373
